### DISCUSSION TITLE
Welcome to moko-mvvm Discussions!

### DISCUSSION THREAD

**Discussion by Alex009** [icerockdev/moko-mvvm#82] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by hakanyildizay** [comment]
Hi @Alex009 
Do you have an example that show emit/listen MediatorLiveData objects.
Thank you.

**Reply by Alex009** [reply]
hi! `MediatorLiveData` can be observed just any other `LiveData` by addObserver
https://github.com/icerockdev/moko-mvvm/blob/master/mvvm-livedata/src/commonMain/kotlin/dev/icerock/moko/mvvm/livedata/LiveData.kt#L10

to emit events you can use `MediatorLiveData.addSource` and combine some other livedata.
https://github.com/icerockdev/moko-mvvm/blob/b6f2630df03bbd405e5659d85ea7df03f38e5dc7/mvvm-livedata/src/commonTest/kotlin/dev/icerock/moko/mvvm/livedata/LiveDataTest.kt#L262 here test with mediator

**Comment by YKakdas** [comment]
Hi @Alex009 ,
When observing livedata, initial value is nofied to observer. Do you consider adding optinal empty constructor to prevent that like androidx.livedata because androidx.livedata has empty constructor without initial value, therefore you may only get notified when the value really has changed, not for the first initialization.
